データベース設計

nyastassia opened the pull request:
**WHAT**

	•	ユーザー、商品、購入、発送先に関する4つのテーブルと、ActiveHashモデルを使用する各属性（カテゴリ・状態・送料負担・発送元・発送日数・都道府県）のカラムを定義した。
	•	itemsテーブルにはenumで管理されるstatus（出品中/売却済）を追加。
	•	クレジットカード情報は外部サービスとのAPI連携により処理し、DBには保存しない設計とした。

**WHY**

	•	ユーザーが商品を出品・購入できるフリマアプリに必要な基本的な機能を実装するため。
	•	データの正規化・セキュリティを考慮し、都道府県やカテゴリなどの静的データはActiveHashで管理する方針とした。
	•	クレジットカード情報はセキュリティリスクを避けるため、保存せず外部APIを通じて処理する設計が一般的なため。

ER図：
[https://gyazo.com/e5c83d1f8d16a91432ff262a42737fa1
![ER](https://github.com/user-attachments/assets/9aff1e57-2d70-48a4-a91f-79ae95b24836)
](url)

nyastassia committed:
データベース設計

wad-sho reviewed (commented):
レビュー依頼確認しました。
以下2点まずはご修正お願いします。

①ER図を修正しましょう。
https://gyazo.com/e5c83d1f8d16a91432ff262a42737fa1
・テーブル以外の情報は不要ですので取り除きましょう。
（users/items/purchases/shipping_addressesテーブルの情報のみ載せましょう）

②READMEを修正しましょう
```
・Activehashは外部キーではないため、外部キー制約を外しましょう。
・カラム名に（）は使用できないため、_を使用しましょう。
・外部キーのカラム名から_idを削除しましょう。
（references型を用いる場合、DB設計では_idの記載が不要になるためです。）
```

修正後、最新のER図のご添付のうえ再度レビュー依頼お願いいたします。

nyastassia committed:
ER図及びread.me 修正_01

①ER図の修正

・テーブル以外の情報の削除

②READMEの修正

・Activehashの外部キー制約を除外
・カラム名の修正。
・references型のカラム名からidを削除

nyastassia commented:
@wad-sho 
レビューありがとうございました。
ご指摘の通り、修正させていただきましたので、ご確認よろしくお願いいたします。

 ER図
[https://gyazo.com/628a23e2906d9bb83130be9c8b44c726](url)

TakashiKozuma commented on README.md:
```diff
@@ -1,24 +1,69 @@
+| item_name        | string     | null: false                               |
+| description      | text       | null: false                               |
+| price            | integer    | null: false                               |
+| status           | integer    | null: false, default: 0                   |
```
```Rb
| status | integer | null: false, default: 0 |
```
商品の状態はcondition_idですでに実装しているので、このカラムは削除しましょう。

TakashiKozuma commented on README.md:
```diff
@@ -1,24 +1,69 @@
+| -----------------| ---------- | ----------------------------------------- |
+| user             | references | null: false, foreign_key: true            |
+| item             | references | null: false, foreign_key: true            |
+| created_at       | date       | null: false                               |
```
```Rb
| created_at       | date       | null: false                               |
```
created_atはデフォルトで備わっているカラムなので、DB設計では削除しておきましょう。

TakashiKozuma commented on README.md:
```diff
@@ -1,24 +1,69 @@
+| ------------- | ---------- | ----------------------------------------- |
+| purchase      | references | null: false, foreign_key: true            |
+| postal_code   | string     | null: false                               |
+| prefecture_id | integer    | null: false                               |
```
```rb
| prefecture_id | integer | null: false |
```
都道府県のカラム名は、商品の都道府県のカラムと揃えましょう。

TakashiKozuma reviewed (commented):
レビュー依頼の内容を確認致しました。
コメントを致しましたので、ご確認お願いします。

修正後はコミット・プッシュのし忘れがないかご確認の上、再度フォームからレビュー依頼をお願い致します。

nyastassia committed:
ER図及びread.me 修正_02

nyastassia commented:
承知しました。
修正のご確認よろしくお願い致します。

 ER図
[https://gyazo.com/069db5956956801bf2b9fa0b9c6bcf08](url)

TakashiKozuma commented:
修正点も含め、確認が完了しました。
LGTMです！！

こちらのブランチは忘れずにMergeいただき、ローカルのMasterにPullした後に次の実装のためのブランチを切ってから次にお進み下さい。

nyastassia commented:
ご確認、ご対応ありがとうございました！
